Hestia/app/src/main/java/hestia/backend/requests/GetRequest.java

```diff
@@ -0,0 +1,68 @@
+package hestia.backend.requests;
+
+import android.util.Log;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+
+/**
+ * This is an abstract class used to do a GET request.
+ * @param <T> the type of the requested object
+ * @see Request
+ */
+
+abstract class GetRequest<T> extends Request {
```

Comment:
We can change T into a JSON object, we know that we will always get a JSON object.